pallets/automation-time/src/lib.rs

```diff
@@ -228,20 +288,57 @@ pub mod pallet {
 			message: Vec<u8>,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
-			if provided_id.len() == 0 {
-				Err(Error::<T>::EmptyProvidedId)?
-			}
 			if message.len() == 0 {
 				Err(Error::<T>::EmptyMessage)?
 			}
-			Self::is_valid_time(time)?;
 
-			let task_id = Self::schedule_task(who.clone(), provided_id.clone(), time)?;
+			Self::validate_and_schedule_task(Action::Notify { message }, who.clone(), provided_id, time)?;
+			Ok(().into())
+		}
 
-			let task = Task::<T>::create_event_task(who.clone(), provided_id, time, message);
-			<Tasks<T>>::insert(task_id, task);
+		/// Schedule a task to fire an event with a custom message.
+		///
+		/// Before the task can be scheduled the task must past validation checks.
+		/// * The transaction is signed
+		/// * The provided_id's length > 0
+		/// * The time is valid
+		/// * 
+		///
+		/// # Parameters
+		/// * `provided_id`: An id provided by the user. This id must be unique for the user.
+		/// * `time`: The unix standard time in seconds for when the task should run.
+		/// * `message`: The message you want the event to have.
+		///
+		/// # Errors
+		/// * `InvalidTime`: Time must end in a whole minute.
+		/// * `PastTime`: Time must be in the future.
+		/// * `DuplicateTask`: There can be no duplicate tasks.
+		/// * `TimeSlotFull`: Time slot is full. No more tasks can be scheduled for this time.
+		/// * `InvalidAmount`: Amount has to be larger than 0.1 OAK.
+		/// * `InvalidRecipient`: Recipient must be a valid address.
+		/// * `TransferToSelf`: Sender cannot transfer money to self.
+		/// * `InsufficientFunds`: Amount in sender account is insufficient.
+		/// * `TransferFailed`: Transfer failed for unknown reason.
+		#[pallet::weight(<T as Config>::WeightInfo::schedule_transfer_task_existing_slot())]
+		pub fn schedule_transfer_task(
+			origin: OriginFor<T>,
+			provided_id: Vec<u8>,
+			time: UnixTime,
+			recipient_id: T:: AccountId,
+			#[pallet::compact] amount: BalanceOf<T>,
```

Comment:
i got it to be Compact<u128>, but it's still using the obscenely large number. need to figure out what substrate is doing with the balance pallet. they dump it right into the transfer function without doing anything else, so i'm not sure why it's not working for me: https://github.com/paritytech/substrate/blob/master/frame/balances/src/lib.rs#L282 

![image](https://user-images.githubusercontent.com/22332240/153982672-e3002563-0e55-41b3-9c54-6e1da5ef78c7.png)

Reply:
note: balance pallet transfer def uses this super large number too